examples/dishwasher-app/silabs/src/ElectricalSensorManager.cpp

```diff
@@ -0,0 +1,251 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/server/Server.h>
+#include <app/util/attribute-storage.h>
+#include <lib/core/ClusterEnums.h>
+
+#include "AppConfig.h"
+#include "AppTask.h"
+#include "DataModelHelper.h"
+#include "DishwasherManager.h"
+#include "ElectricalEnergyMeasurementInstance.h"
+#include "ElectricalSensorManager.h"
+#include "EnergyTimeUtils.h"
+#include "PowerTopologyDelegate.h"
+#include <ElectricalPowerMeasurementDelegate.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::ElectricalEnergyMeasurement;
+using namespace chip::app::Clusters::ElectricalEnergyMeasurement::Structs;
+using namespace chip::app::Clusters::ElectricalPowerMeasurement;
+using namespace chip::app::Clusters::DeviceEnergyManagement;
+using namespace chip::app::Clusters::OperationalState;
+using namespace chip::app::Clusters::PowerTopology;
+using namespace chip::app::DataModel;
+
+ElectricalSensorManager ElectricalSensorManager::sElectricalSensor;
+
+static std::unique_ptr<ElectricalEnergyMeasurementInstance> gEEMInstance;
+static std::unique_ptr<ElectricalPowerMeasurementDelegate> gEPMDelegate;
+static std::unique_ptr<ElectricalPowerMeasurementInstance> gEPMInstance;
+static std::unique_ptr<PowerTopologyDelegate> gPTDelegate;
+static std::unique_ptr<PowerTopologyInstance> gPTInstance;
+
+static const struct
+{
+    PowerModeEnum PowerMode;
+    int64_t Voltage;         // mV
+    int64_t ActiveCurrent;   // mA
+    int64_t ReactiveCurrent; // mA
```

Comment:
I would question if you really need all of these optional measurements for a real device. Voltage, ActiveCurrent, ActivePower and maybe Frequency should be enough for a white goods appliance? But nothing to stop you having them as examples I suppose.